core/src/scene/sceneLoader.cpp

```diff
@@ -896,11 +899,34 @@ void SceneLoader::loadSource(const std::string& name, const Node& source, const
     }
 
     std::string type = source["type"].Scalar();
-    std::string url = source["url"].Scalar();
+    std::string url;
+    std::string mbtiles;
     int32_t minDisplayZoom = -1;
     int32_t maxDisplayZoom = -1;
     int32_t maxZoom = 18;
 
+    std::string mime;
+
+    if (type == "GeoJSON") {
```

Comment:
Perhaps we can discern the mime type automatically from the `RawDataSoure` by just returning the mime type of the `DataSource`? We will know the mime type of the `DataSource` based on the implementation of the `DataSource`. Then, you can put the logic of figuring out the mime type for raster sources in `RasterSource` instead of in here?